code/modules/mob/living/simple_animal/bot/bot.dm

```diff
@@ -948,6 +962,29 @@ Pass a positive integer as an argument to override a bot's default speed.
 	. = ..()
 	icon_state = "[initial(icon_state)][on]"
 
+
+/mob/living/simple_animal/bot/MouseDrop(over_object, src_location, over_location)
+	. = ..()
+	if(over_object == usr && Adjacent(usr))
+		if(!ishuman(usr) || !usr.canUseTopic(src, BE_CLOSE))
+			return FALSE
+		if(!carryable)
+			to_chat(usr, "<span class='notice'>[src] too large to carry!</span>")
+			return FALSE
+		if(on || booting)
+			to_chat(usr, "<span class='notice'>You need to turn [src] off before carrying it around.</span>")
+			return FALSE
+		usr.visible_message("<span class='notice'>[usr] picks up the [src].</span>", "<span class='notice'>You pick up [src].</span>")
+		var/obj/item/deployable/bot/carried = new(loc)
+		carried.deployed_object = src
```

Comment:
This has the potential to result in a hard-del of this mob is somehow deleted while inside of the object. Also it may poorly deal with the mob managing to escape the deployed bot object, since that could happen due to things like xeno extracts.
Rather than store a reference to deployed_object, you could just drop all objects in contents instead. You'll have to figure out Exited() as well in order to delete the deployable bot object if the mob somehow escapes the deployable via forceMove.

Reply:
This is now resolved in the new PR. Shuffled around exactly how the bot is deployed a bit and also added this as a self-destruct sequence of sorts:

```
/obj/item/deployable/bot/Initialize(mapload)
    RegisterSignal(src, COMSIG_ATOM_EXITED, PROC_REF(self_destruct))

/obj/item/deployable/bot/proc/self_destruct()
    SIGNAL_HANDLER
    for(var/atom/movable/A in contents)
        return
    qdel(src)
```